src/pages/studyView/addChartButton/addChartByType/AddChartByType.tsx

```diff
@@ -69,83 +85,112 @@ export default class AddChartByType extends React.Component<IAddChartByTypeProps
+        name: 'Freq',
+        tooltip: <span>% samples with data</span>,
         render: (option: ChartOption) =>
-            <span className={classnames(option.disabled ? styles.labelDisabled : '')}>
+            <span style={{display: 'flex', flexDirection: 'row-reverse'}}
+                  className={classnames(option.disabled ? styles.labelDisabled : '')}>
                 {this.props.freqPromise.isComplete ? getFrequencyStr(option.freq) : ''}
             </span>,
         sortBy: (d: ChartOption) => d.freq,
+        headerRender: () => {
+            return <span style={{display: 'flex', flexDirection: 'row-reverse', flexGrow: 1}}>Freq</span>
+        },
         defaultSortDirection: 'desc' as 'desc',
-        width: 160
+        width: 60
     }];
 
     @computed
     get tableHeight() {
-        return this.options.length > NUM_ROWS_SHOWN ? NUM_ROWS_SHOWN * 25 : this.options.length * 25;
+        return this.options.length > NUM_ROWS_SHOWN ? NUM_ROWS_SHOWN * 25 : (this.options.length + 1) * 25;
+    }
+
+    @autobind
+    getCurrentSelectedRows() {
+        return _.filter(this.options, option => option.selected);
+    }
+
+    @autobind
+    getCurrentSelectedRowKeys(){
+        return this.getCurrentSelectedRows().map(option => option.key);
     }
 
     @autobind
     @action
     addAll(selectedOptions: ChartOption[]) {
-        this.props.onAddAll(_.uniq(_.filter(this.options, option=>option.selected).concat(_.filter(selectedOptions, option=>!option.disabled))).map(option => option.key));
+        const alreadySelected = this.getCurrentSelectedRows();
+        const allSelected = _.filter(selectedOptions, option => !option.disabled);
+        const numOfNewCharts = allSelected.length - alreadySelected.length;
+        this.props.onAddAll(_.uniq(alreadySelected.concat(allSelected).map(option => option.key)));
+        this.infoMessage = `${numOfNewCharts} chart${numOfNewCharts > 1 ? 's' : ''} added`;
     }
 
     @autobind
     @action
     removeAll(selectedOptions: ChartOption[]) {
+        const alreadySelected = this.getCurrentSelectedRowKeys();
+        const allSelected = selectedOptions.map(option => option.key);
+        const numOfChartsRemoved = _.intersection(alreadySelected, allSelected).length;
         this.props.onClearAll(selectedOptions.map(option => option.key));
+        this.infoMessage = `${numOfChartsRemoved} chart${numOfChartsRemoved > 1 ? 's' : ''} removed`;
+    }
+
+    @autobind
+    @action
+    onOptionChange(option: ChartOption) {
+        const selectedKeys = this.getCurrentSelectedRowKeys();
+        this.props.onToggleOption(option.key)
+        if (selectedKeys.includes(option.key)) {
+            this.infoMessage = `${option.label} is deleted`;
+        }else {
+            this.infoMessage = `${option.label} added as a ${ChartTypeNameEnum[option.chartType]}`;
+        }
+    }
+
+    componentWillUnmount(): void {
+        this.reactions.forEach(reaction => reaction());
```

Comment:
Is this to dispose reactions on unmount?

Reply:
yes